src/lesson-1-9-demo/SignupForm.jsx

```diff
@@ -1,23 +1,165 @@
 import { useEffect, useState } from "react";
 import "./signup-form.css";
 
+const createUser = async (userData) => {
+  try {
+    const response = await fetch("http://localhost:3001/api/users", {
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      },
+      method: "POST",
+      body: JSON.stringify(userData),
+    });
+
+    if (response.ok) {
+      return response.json();
+    }
+
+    throw Error("Failed to create user");
+  } catch (error) {
+    console.error({ error });
+    throw error;
+  }
+};
+
+const initialFormState = {
+  username: "",
+  email: "",
+  password: "",
+  passwordConfirmation: "",
+};
+
 const SignupForm = () => {
+  const [formState, setFormState] = useState(initialFormState);
+  const [formErrors, setFormErrors] = useState({});
+
+  const validations = [
```

Comment:
here we define a list of validations to run through before w can submit the form